Staged/Classes/STGNavigationControllerVerifier.m

```diff
@@ -0,0 +1,105 @@
+    self = [super init];
+    if (self) {
+        _viewControllers = @[];
+
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(viewControllerWasPushed:)
+                                                     name:STGViewControllerWasPushedNotificationName
+                                                   object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(viewControllerWasPopped:)
+                                                     name:STGViewControllerWasPoppedNotificationName
+                                                   object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(navigationWasPoppedToRoot:)
+                                                     name:STGNavigationControllerWasPoppedToRootNotificationName
+                                                   object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(navigationWasPoppedToViewController:)
+                                                     name:STGNavigationControllerWasPoppedToViewControllerNotificationName
+                                                   object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(navigationWasReset:)
+                                                     name:STGNavigationControllerItemsWereSetNotificationName
+                                                   object:nil];
+
+        swizzleMocks();
+    }
+
+    return self;
+}
+
+- (void)dealloc {
+    swizzleMocks();
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (UIViewController  * _Nullable )topViewController {
+    return _viewControllers.lastObject;
+}
+
+- (void)viewControllerWasPushed:(NSNotification *)notification {
+    NSMutableArray *viewControllers = [self.viewControllers mutableCopy];
+    [viewControllers addObject:notification.object];
+    self.viewControllers = viewControllers;
+    self.pushedAnimated = [notification.userInfo[STGViewControllerPushingAnimatedKey] boolValue];
+}
+
+- (void)viewControllerWasPopped:(NSNotification *)notification {
+    NSMutableArray *viewControllers = [self.viewControllers mutableCopy];
+    [viewControllers removeLastObject];
+    self.viewControllers = viewControllers;
+    self.poppedAnimated = [notification.userInfo[STGViewControllerPoppingAnimatedKey] boolValue];
+}
+
+- (void)navigationWasPoppedToRoot:(NSNotification *)notification {
+    NSMutableArray *viewControllers = [self.viewControllers mutableCopy];
+    while (viewControllers.count > 1) {
+        [viewControllers removeLastObject];
+    }
+    self.viewControllers = viewControllers;
```

Comment:
Yeah, I misread that 😅 

What about `self.viewControllers = @[self.viewControllers.firstObject]`?